xscope_fileio/src/xscope_io_device.c

```diff
@@ -110,18 +110,21 @@ size_t xscope_fread(xscope_file_t *xscope_file, uint8_t *buffer, size_t n_bytes_
         {
         read_host_data:
             {
-                xscope_data_from_host(c_xscope, (char *)buffer_ptr, &bytes_read);
-                // printf("rx %u bytes\n", bytes_read);
-                end_marker_found = ((bytes_read == END_MARKER_LEN) && !memcmp(buffer_ptr, END_MARKER_STRING, END_MARKER_LEN)) ? 1 : 0;
+                // need a buffer big enough to hold max read length, user provided
+                // buffer probably isn't that size.
+                char local_buffer[MAX_XSCOPE_SIZE_BYTES];
```

Comment:
This is sensible. I'd change the comment to say user provided buffer may be smaller.